### PR TITLE
Doorslug/Breaching slug/Door breaching shotgun round

### PR DESCRIPTION
**Pull request by LanCartwright**
this breaks doors for your tacticool nukeop goodness.

chamber a doorbuster, slot in some buckshot, breach a room and have 8 rounds of triple aught buckshot to you

maximum tacticool. will add this to nukeop later. this is proof of concept

### CODE REVIEW

**Comment by Lumipharon** [comment]
So this is meteor shot light? Fucking excellent. Add to ops post haste.

**Comment by LanCartwright** [comment]
gotta consult it with the elders first and gather some opinions.

File: code/modules/projectiles/ammunition/ammo_casings.dm [hunk 1]
```diff
@@ -142,6 +142,12 @@
 	icon_state = "mshell"
 	projectile_type = /obj/item/projectile/bullet/meteorshot
 
+/obj/item/ammo_casing/shotgun/doorslug
+	name = "breaching slug"
+	desc = "A modified shotgun shell which fires a massive non-neutonian fluid projectile when fired. Extremely useful for breaching doors. Not useful for breaching rigcages."
```
Comment (tkdrg) [inline code/modules/projectiles/ammunition/ammo_casings.dm]: Is this supposed to be `ribcages`?

File: code/modules/projectiles/projectile/bullets.dm [hunk 2]
```diff
@@ -112,6 +112,25 @@
...
+	weaken = 2
+	stun = 2
+	hitsound = 'sound/effects/meteorimpact.ogg'
+
+/obj/item/projectile/bullet/doorslug/on_hit(atom/target, blocked = 0)
+	. = ..()
+	if(istype(target, /atom/movable))
+		var/atom/movable/M = target
+		var/atom/throw_target = get_edge_target_turf(M, get_dir(src, get_step_away(M, src)))
+		M.throw_at(throw_target, 2, 1)
```
Comment (tkdrg) [inline code/modules/projectiles/projectile/bullets.dm]: You sure that this works with anchored stuff like doors?

Comment (RemieRichards) [inline code/modules/projectiles/projectile/bullets.dm]: pretty sure throw_at cares not for anchored.

Comment (LanCartwright) [inline code/modules/projectiles/projectile/bullets.dm]: I could have sworn I've broken a door down once with this, hence why I jumped on coding it so quickly. Could a quick check be run on this?

**Comment by Lumipharon** [comment]
meteor shot works for doors, so check how it does it.

**Comment by Steelpoint** [comment]
I like the idea but it further pushes you to never want to use anything aside from the Bulldog.

Why not? You get a powerful gun that can blow open doors!?!?

Honestly I think going with a free/cheap semtex charge that just instantly blows up a door only would be nice, since that's independent of any weapon.

**Comment by Lumipharon** [comment]
with the new op gun changes, slugs should just get removed from bulldogs. Leave it as a utility weapon, with it current various rounds, plus extra shit like emp slugs etc etc.

**Comment by LanCartwright** [comment]
Changelog:

Added to uplink:
Bulldog drum magazine - 12 gauge door slugs. 5 TC.

Added to boxes:
Door breaching slugs - contains 7 door slugs.

Adds recipe for door slugs:
1 Empty tech shell.
10 flour.
10 water.

Added door slug casing.
Added door slug Bulldog magazine

Added doorslug projectile;
10 damage. 2 Weaken. 2 stun. Knocks doors back.

Adds new sprites for doorslug boxes, door slugs used and live, and doorslug drum magazines full and empty.

**Comment by Steelpoint** [comment]
Remove free bulldog when?

**Comment by LanCartwright** [comment]
oh sh- doing that

**Comment by tkdrg** [comment]
Might be easier to just break down the door to avoid the "tactical greyshirt in front of door" issue.

**Comment by Lumipharon** [comment]
Lan, people can already make meteor shot with tech shells, which are objectively just better.
Also I think making slugs with FLOUR and WATER is a tad retarded.

**Comment by LanCartwright** [comment]
I call it a feature. If you know someone is gonna blow the hinges off your door, you stick something behind it to prevent it.

They teach that to teachers in US schools, and security officers as to what to do when a threat attempts to enter a room.

**Comment by Lumipharon** [comment]
Yeah but like, you can put ANYTHING behind a door, and it makes meteor shot/door breacher utterly fucking useless.

Like, standing behind it/putting ian there/a locker is a HARD COUNTER.

File: code/modules/projectiles/projectile/bullets.dm [hunk 3]
```diff
@@ -112,6 +112,27 @@
...
+	weaken = 2
+	stun = 2
+	hitsound = 'sound/effects/meteorimpact.ogg'
+
+/obj/item/projectile/bullet/doorslug/on_hit(atom/target, blocked = 0)
+	. = ..()
+	if(istype(target, /atom/movable))
+		var/atom/movable/M = target
+		var/atom/throw_target = get_edge_target_turf(M, get_dir(src, get_step_away(M, src)))
+		var/atom/target_anchored = 0
```
Comment (octareenroon91) [inline code/modules/projectiles/projectile/bullets.dm]: You only need ```target.anchored = 0``` on this line.  ```var/``` stuff is for defining a new variable that didn't exist before.

File: code/modules/projectiles/projectile/bullets.dm [hunk 4]
```diff
@@ -112,6 +112,27 @@
...
+		M.throw_at(throw_target, 2, 1)
+		var/atom/target_anchored = 1
```
Comment (octareenroon91) [inline code/modules/projectiles/projectile/bullets.dm]: You don't need var/ except for new variables.
```
var/temp_anchored = target.anchored
M.throw_at(throw_target, 2, 1)
target.anchored = temp_anchored
```

**Comment by Cheridan** [comment]
stale, also this is basically meteorshot yeah

**Comment by Lumipharon** [comment]
I'm fine for closing this, but not for the above reason.

It's similar, but not identical to meteor shot, because meteor shot would be rediculously OP in a bulldog (stun + 20 brute + the push back).

The actual issue with this pr is that it's intended purpose of getting through airlocks even when bolted, is hard countered by putting anything on the other side of the airlock to stop it moving, since meteor shot doesn't unanchor the object, just move it.

The pr needs to be redone to unanchor airlocks when hit, or simply blow the airlock apart.

**Comment by octareenroon91** [comment]
What if the bullet could unanchor and throw literally anything?

**Pull request closed without merging**